pnet/Private-Networks-PSK-V1.md

```diff
@@ -0,0 +1,75 @@
+## Pre-shared Key Based Private Networks in IPFS
+
+This document describes the first version of private networks (PN) featured in IPFS.
+
+For the first implementation, only pre-shared key (PSK) functionality is available, as the Public Key Infrastructure approach is much more complex and requires more technical preparation.
+
+It was implemented as an additional encryption layer before any IPFS traffic, and was designed to leak the absolute minimum of information on its own. All traffic leaving the node inside a PN is encrypted and there is no characteristic handshake.
+
+### Interface
+
+An IPFS node or libp2p swarm is either in a a public network, or it is a member of a private network.
+
+A private network is defined by the 256-bit secret key, which has to be known and used by all members inside the network.
+
+In the case of an IPFS node, this key is stored inside the IPFS repo in a file named `swarm.key`. The file uses a path-based multicodec where, for now, the codec that is defined and used is `/key/swarm/psk/1.0.0/`. The codec expects the next path-based multicodec to define the base encoding for the rest of the file (`/bin/`, `/base16/`, `/base64/`), which is the 256-bit PSK. The key has to be exactly 256-bits (32 bytes) long.
+
+#### Security Guarantees
+
+Nodes of different private networks must not be able to connect to each other. This extends to node in private network connecting to node in public network. This means that no information exchange, apart from the handshakeing required for private network authentication, should take place.
```

Comment:
`must not be able` suggests that there is no strict guarantee, which comes off in the 'Security Guarantees' section, perhaps 'Security Considerations?'